### PR TITLE
restrictions code review

### PR DESCRIPTION
**Pull request by Snover98**
These classes are for checking if values qualify for the restrictions defined in the XML schema

### CODE REVIEW

File: Restrictions/RangeInclusiveRestriction.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    def value_matches_restriction(self, value: str) -> bool:
+        try:
+            value_converted = self.parent_restriction(value)
+        except RestrictionException as e:
+            return False
+
+        if self.minimum is None:
+            return value_converted <= self.maximum
+        elif self.maximum is None:
+            return self.maximum <= value_converted
```
Comment (NivSikourel) [inline Restrictions/RangeInclusiveRestriction.py]: i think it should be return self.minimum <= value_converted. No?

File: Restrictions/IntegerRestrictions.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+class ByteIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = -128
+    MaxRangeEdge = 127
+
+    def __init__(self, minimum: int = MinRangeEdge, maximum: int = MaxRangeEdge):
+        super().__init__("byte", minimum, maximum)
+
+
+class UnsignedLongIntegerRestriction(NonNegativeIntegerRestriction):
+    MaxRangeEdge = 18446744073709551615
```
Comment (NivSikourel) [inline Restrictions/IntegerRestrictions.py:121]: maybe add MinRangeEdge = 0?

File: Restrictions/IntegerRestrictions.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    def __init__(self, maximum: int = MaxRangeEdge):
+        super().__init__("unsignedLong", maximum)
+
+
+class UnsignedIntIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MaxRangeEdge = 4294967295
```
Comment (NivSikourel) [inline Restrictions/IntegerRestrictions.py:129]: maybe add MinRangeEdge = 0?

File: Restrictions/IntegerRestrictions.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    def __init__(self, maximum: int = MaxRangeEdge):
+        super().__init__("unsignedInt", maximum)
+
+
+class UnsignedShortIntegerRestriction(NonNegativeIntegerRestriction):
+    MaxRangeEdge = 65535
```
Comment (NivSikourel) [inline Restrictions/IntegerRestrictions.py:137]: maybe add MinRangeEdge = 0?

File: Restrictions/IntegerRestrictions.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+    def __init__(self, maximum: int = MaxRangeEdge):
+        super().__init__("unsignedShort", maximum)
+
+
+class UnsignedByteIntegerRestriction(NonNegativeIntegerRestriction):
+    MaxRangeEdge = 255
```
Comment (NivSikourel) [inline Restrictions/IntegerRestrictions.py:145]: maybe add MinRangeEdge = 0?

**Review by NivSikourel — CHANGES_REQUESTED** [review]
Very good job overall. I left some comments for you to have fun with :)

File: Restrictions/RangeInclusiveRestriction.py (hunk 1 above)
Comment (Snover98) [reply Restrictions/RangeInclusiveRestriction.py]: Your'e absolutely right

File: Restrictions/IntegerRestrictions.py (hunk 2 above)
Comment (Snover98) [reply Restrictions/IntegerRestrictions.py:121]: already done in the base class NonNegativeIntegerRestriction

**Pull request merged**